### DISCUSSION TITLE
Do you plan to maintain this repo for a long run?

### DISCUSSION THREAD

**Discussion by mtroym** [lucafaggianelli/plombery#393] [Q&A]
I want or I will implement some of the functionalities:
1. cancel any run from ui
2. automatically trigger new job when current job running. (self trigger)

**Comment by lucafaggianelli** [comment]
Hey @mtroym, thanks for your interest in Plombery! So the tl;dr answer is **yes**, I plan to maintain this repo!

I planned to restart working on in it after summer, end of August / beginning of September, the idea is to work on bugs and make the platform stable and then I want to understand what direction to go with the new features, so I planned to have some interviews or surveys with actual users.

Said so, please feel free to work on issue #285, though consider that in [APScheduler v4](https://github.com/agronholm/apscheduler/issues/465) they're planning to add this functionality

About 2. I have mixed feelings / solutions:
- in issue #342 where the user wants to refresh oauth2 secrets, probably we can think about Credentials plugins for Plombery, where the user saves their own credentials (api key, oauth, etc.) and they're injected into Plombery's Tasks (indeed this could be a direction for Plombery: creating 3rd parties integrations)
- we can introduce conditional tasks, so one add a task to the pipeline but there's a pre-condition to check before running it (i.e. a Python function that returns True/False) and that task could be skipped or the entire pipeline stopped
- if we make the Task objects callable #156 one could call a Task from another Task
